server/src/main/java/org/elasticsearch/index/codec/CodecService.java

```diff
@@ -120,4 +152,16 @@ public final Codec delegate() {
         }
 
     }
+
+    private enum StorageMode {
```

Comment:
I find this storage mode a bit confusing. Maybe a `useBloomFilterSyntheticId` local variable would be simpler?

Reply:
I got rid of it in https://github.com/elastic/elasticsearch/pull/138515/commits/a72a66a7475189873c0a4624928a6f9fb419e882